File: old/apps/backend/src/modules/buildpack-version/buildpack-version.entity.ts
import {
  Entity,
  ManyToOne,
  OneToOne,
  PrimaryKey,
  Property,
} from '@mikro-orm/core';
import { v4 } from 'uuid';
import { Buildpack } from '../buildpack/buildpack.entity';

@Entity({ tableName: 'buildpack_versions' })
export class BuildpackVersion {
  @PrimaryKey()
  id: string = v4();

  @Property()
  version: string;

  @ManyToOne(() => Buildpack)
  buildpack: Buildpack;

  @Property({ onCreate: () => new Date() })
  creationDate: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  modificationDate: Date = new Date();
}

File: backend/src/main/java/com/esmo/empaas/entities/BuildpackVersionEntity.java
package com.esmo.empaas.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.util.Date;

@Entity
@Table(name = "buildpack_versions")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class BuildpackVersionEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(name = "version")
    private String version;

    @ManyToOne
    @JoinColumn(name = "buildpack_id")
    private BuildpackEntity buildpack;

    @CreationTimestamp
    @Column(name = "creation_date")
    private Date creationDate;

    @UpdateTimestamp
    @Column(name = "modification_date")
    private Date modificationDate;

}
